bundles/org.openhab.binding.clementineremote/src/main/java/org/openhab/binding/clementineremote/internal/ClementineRemoteHandler.java

```diff
@@ -0,0 +1,425 @@
+        }
+
+        updateState(CHANNEL_POSITION, new QuantityType<>(currentPos, Units.SECOND));
+    }
+
+    private void handleTrackPos(ResponseUpdateTrackPosition message) {
+        handleNullableTrackPos(message.getPosition());
+    }
+
+    private void handleVolume(RequestSetVolume message) {
+        volume = message.getVolume();
+        updateState(CHANNEL_VOLUME, new PercentType(volume));
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(ClementineRemoteConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+        scheduler.execute(this::connect);
+    }
+
+    private void sleep15secs() {
+        try {
+            sleep(15000);
+        } catch (InterruptedException e) {
+            logger.warn("sleep interrupted: {}", e.getMessage());
+        }
+    }
+
+    private void sendConnectMessage() throws IOException {
+        var req = builder.getRequestConnectBuilder();
+        if (config.authCode != null && config.authCode != 0) {
+            req.setAuthCode(config.authCode);
+        }
+        sendMessageOrThrow(builder.setRequestConnect(req.build()).build());
+    }
+
+    private boolean sendSkip(int d) {
+        state = State.SKIPPING;
+        var pos = RequestSetTrackPosition.newBuilder().setPosition(currentPos + d);
+        Message msg = builder.setType(MsgType.SET_TRACK_POSITION).setRequestSetTrackPosition(pos).build();
+        return sendMessage(msg);
+    }
+
+    private boolean sendMessage(MsgType type) {
+        state = State.UNKNOWN;
+        return sendMessage(builder.setType(type).build());
+    }
+
+    private boolean sendMessage(Message message) {
+        if (out == null) {
+            return false;
+        }
+        try {
+            sendMessageOrThrow(message);
+            return true;
+        } catch (IOException e) {
+            logger.warn("sendMessage({}) failed: {}", message, e.getMessage());
+            disconnect();
+            return false;
+        }
+    }
+
+    private void sendMessageOrThrow(Message message) throws IOException {
+        byte[] bytes = message.toByteArray();
+        out.writeInt(bytes.length);
+        out.write(bytes);
+        out.flush();
+    }
+
+    private boolean sendVolume(int percent) {
+        var volume = RequestSetVolume.newBuilder().setVolume(percent);
+        Message msg = builder.setType(MsgType.SET_VOLUME).setRequestSetVolume(volume).build();
+        return sendMessage(msg);
+    }
+
+    private boolean setState(State newState) {
+        if (state == newState) {
+            return false;
+        }
+        updatePlayerStateChannels(state = newState);
+        return true;
+    }
+
+    @Override
+    protected void updateConfiguration(Configuration configuration) {
+        super.updateConfiguration(configuration);
+        dispose();
+        scheduler.execute(this::connect);
+    }
+
+    private void updatePlayerStateChannels(State state) {
+        switch (state) {
```

Comment:
`SKIPPING` and `Unknown` are missing. I think you want to list them and intentionally ignore them.

Reply:
Fixed. I dropped the whole state channel, which was redundant with playback channel.